connectors/src/connectors/zendesk/temporal/cast_known_errors.ts

```diff
@@ -0,0 +1,48 @@
+import type {
+  ActivityExecuteInput,
+  ActivityInboundCallsInterceptor,
+  Next,
+} from "@temporalio/worker";
+
+import {
+  isNodeZendeskForbiddenError,
+  isZendeskEpipeError,
+  isZendeskExpiredCursorError,
+} from "@connectors/connectors/zendesk/lib/errors";
+import {
+  ExternalOAuthTokenError,
+  ProviderWorkflowError,
+} from "@connectors/lib/error";
+
+export class ZendeskCastKnownErrorsInterceptor
+  implements ActivityInboundCallsInterceptor
+{
+  async execute(
+    input: ActivityExecuteInput,
+    next: Next<ActivityInboundCallsInterceptor, "execute">
+  ): Promise<unknown> {
+    try {
+      return await next(input);
+    } catch (err: unknown) {
+      if (isNodeZendeskForbiddenError(err)) {
+        throw new ExternalOAuthTokenError(err);
+      } else if (isZendeskExpiredCursorError(err)) {
+        throw new ProviderWorkflowError(
+          "zendesk",
+          "Cursor expired",
+          "transient_upstream_activity_error",
```

Comment:
Do we think this one will be transient ?

Reply:
actually this one is not transient, the activity here is part of a whole bunch of paginated activities that is transient as a whole

Reply:
should I just put a `unhandled_internal_activity_error`?

Reply:
Yeah because the retry is at the page level no ? So if we end up with this error we're stuc,

Reply:
understood thanks 👍

Reply:
fixed